src/Avalonia.Controls/Primitives/SelectingItemsControl.cs

```diff
@@ -1118,6 +1114,26 @@ private void UpdateFinished()
             }
         }
 
+        private void InternalBeginInit()
+        {
+            if (_updateCount == 0)
```

Comment:
I think this is actually expected behavior, without this check if user would `BeginInit` -> select something -> `BeginInit` we would reset their changes.